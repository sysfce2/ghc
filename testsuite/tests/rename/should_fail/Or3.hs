{-# LANGUAGE OrPatterns #-}

module Main where

f x = case x of
  (Left a | Right a) -> a

g x = case x of
  (_ | (x | _)) -> x

h x = case x of
  (Just @Int 3 | Nothing) -> 2

main = print $ foo 3 IsInt1