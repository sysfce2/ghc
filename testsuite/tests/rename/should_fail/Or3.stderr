
Or3.hs:6:3: error: [GHC-81303]
    An or-pattern may not bind (type) variables nor type class or equality constraints: (Left a
                                                                                         | Right a)

Or3.hs:9:8: error: [GHC-81303]
    An or-pattern may not bind (type) variables nor type class or equality constraints: (x
                                                                                         | _)

Or3.hs:12:3: error: [GHC-28418]
    An or-pattern may not contain visible type applications: (Just @Int
                                                                   3
                                                              | Nothing)
